src/GiraffeGenerator.Core/OpenApiToAstTypesMatchingAndConversion.fs

```diff
@@ -49,30 +49,63 @@ let strFormatMatch =
     | Custom "uid" -> guidType
     | Custom name -> synType name
 
+let strFormatNodaTimeDateTimeMatch =
+    function
+    | DateTimeGeneratedType.ZonedDateTime -> nodaTypes.ZonedDateTime
+    | DateTimeGeneratedType.OffsetDateTime -> nodaTypes.OffsetDateTime
+    | DateTimeGeneratedType.LocalDateTime -> nodaTypes.LocalDateTime
+    | DateTimeGeneratedType.Instant -> nodaTypes.Instant
+
+let strFormatNodaTimeMatch config =
+    function
+    | Custom "local-date"
+    | DateString -> Some nodaTypes.LocalDate
+    | DateTimeString -> strFormatNodaTimeDateTimeMatch config.MapDateTimeInto |> Some
+    | Custom "instant" -> Some nodaTypes.Instant
+    | Custom "time"
+    | Custom "local-time" -> Some nodaTypes.LocalTime
+    | Custom "local-date-time" -> Some nodaTypes.LocalDateTime
+    | Custom "offset-date-time" -> Some nodaTypes.OffsetDateTime
+    | Custom "zoned-date-time" -> Some nodaTypes.ZonedDateTime
+    | Custom "offset"
+    | Custom "time-offset" -> Some nodaTypes.Offset
+    | Custom "duration" -> Some nodaTypes.Duration
+    | Custom "period" -> Some nodaTypes.Period
+    | Custom "time-zone"
+    | Custom "date-time-zone" -> Some nodaTypes.DateTimeZone
+    | _ -> None
+    
+let strFormatMatch config format =
+    Some config
+    |> Option.filter ^ fun x -> x.UseNodaTime
+    |> Option.map strFormatNodaTimeMatch
+    |> Option.bind ^ (|>) format
+    |> Option.defaultWith ^ fun _ -> strFormatDefaultMatch format
+
 /// matching OpenAPI primitive type IR to SynType
-let primTypeMatch =
+let primTypeMatch config =
```

Comment:
this config is being passed everywhere.
Maybe instead of polluting code with passing config explicitly let's create global mutable config? :)

Reply:
Done